lib/diyepw/diyepw/get_noaa_isd_lite_file.py

```diff
@@ -0,0 +1,90 @@
+import urllib.request as _request
+import re as _re
+import os as _os
+import pandas as _pd
+
+def get_noaa_isd_lite_file(wmo_index:int, year:int, output_dir:str = None, force_update = False) -> str:
+    """
+    Given a WMO index and a year, retrieve the corresponding NOAA ISD Lite AMY file
+    :param wmo_index:
+    :param year:
+    :param output_dir: Optional output directory - if not specified, the file will be saved to a package directory.
+        If the directory already contains a NOAA ISD Lite file matching the requested WMO Index and year, then a new
+        file will not be downloaded from NOAA and that file's path will be returned
+    :param force_update: Pass True to force a new ISD Lite file to be downloaded, even if it already exists in the
+       output directory.
+    :return: The path to the NOAA ISD Lite file
+    """
+    if output_dir is None:
+        this_dir = _os.path.dirname(_os.path.realpath(__file__))
```

Comment:
it is considered bad form to use `__file__` inside a package.  Use methods from `pkg_resources` instead

Reply:
I've created https://github.com/IMMM-SFA/diyepw/issues/29 to track this, and I'll implement it in the csv_based_script branch. I don't want to do it in this branch because there have been so many changes to the path handling that it's going to be a lot faster to just do it at the current development head instead of doing it in this branch and trying to merge it.